nixos/modules/misc/documentation.nix

```diff
@@ -267,7 +267,7 @@ in
       system.build.manual = manual;
 
       environment.systemPackages = []
-        ++ optional cfg.man.enable manual.manpages
+        ++ optional (cfg.man.enable || cfg.mandoc.enable) manual.manpages
```

Comment:
(Let `cfg = config.documentation`.) If practical, I'd like to see the utility-specific (/ implementation-specific) configuration for `pkgs.man-db` and `pkgs.mandoc` be factored out to attributes under `cfg.man.man-db` and `cfg.man.mandoc`, to allow for a generic control for manual pages & tooling. The `pkgs.man-db` configuration in `modules/misc/documentation.nix` should be factored out to `modules/misc/man-db.nix`, like `modules/misc/mandoc.nix`.

`cfg.man.enable` can continue to install `pkgs.man-db` by default, but that should be via setting a default `cfg.man.man-db.enable`. mandoc's specific config should live at `cfg.man.mandoc`. Generic options should continue to exist at the top-level if possible and be used as defaults, like for `pkgs.man-db` a `config = lib.mkIf cfg.man.enable { documentation.man.man-db.generateCaches = lib.mkDefault cfg.man.generateCaches; };`, and for `pkgs.mandoc` a `config = lib.mkIf cfg.man.enable { documentation.man.mandoc.generateCaches = lib.mkDefault cfg.man.generateCaches; }`.

This keeps things cleaner & more consistent, even if only man-db and mandoc end up being used. I'm worried about `cfg.man.enable` not being a reliable option for other modules to check, and that causing difficult-to-remove messiness elsewhere. It also clarifies the separation between installation of manual pages from packages into the environment and installation of `pkgs.man-db` into the environment.